### ISSUE TITLE
Consider renaming connect/disconnect to link/unlink

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#5327]
## Problem

Right now the terms `connect` and `disconnect` are overloaded in the Prisma client

### Connect

- `prisma.$connect`: connect to a database
- nested `connect`: connect a relation

### Disconnect

- `prisma.$disconnect`: disconnect to a database
- nested `disconnect`: disconnect a relation

## Suggested solution

Deprecate nested `connect` and `disconnect`, replace them with `link` and `unlink`.

```diff
const user = await prisma.profile.create({
  data: {
    bio: 'Hello World',
    user: {
-     connect: { email: 'alice@prisma.io' },
+     link: { email: 'alice@prisma.io' },
    },
  },
})
```

There would be a period of deprecation and the codemod would be updated to help with the transition.

## Alternatives

Do nothing. Decide that it's not that confusing. I'd love to hear what you think 🙂

**Comment by janpio** [comment]
Previously: https://github.com/prisma/prisma/issues/4360 https://github.com/prisma/prisma-client-js/issues/348

**Comment by steebchen** [comment]
I agree we should change this. The Go client already uses Link and Unlink and I think it's a lot clearer at what it does.
